traveler-ios-ui/TravelerKitUI/ViewControllers/ParkingQueryFormViewController.swift

```diff
@@ -0,0 +1,194 @@
+             // pickup date cell
+             let cell = tableView.dequeueReusableCell(withIdentifier: datePickerCellIdentifier, for: indexPath) as! DatePickerCell
+             cell.datePicker.minimumDate = query!.dateRange.lowerBound
+             cell.datePicker.date = query!.dateRange.upperBound
+             cell.delegate = self
+             cell.tag = pickUpTag
+             return cell
+        default:
+            fatalError("Invalid indexPath")
+        }
+    }
+
+    @IBAction func didUpdate(_ sender: Any) {
+        locationManager?.stopUpdatingLocation()
+
+        guard let query = query else {
+            return
+        }
+
+        delegate?.parkingQueryFormViewController(self, didUpdate: query)
+    }
+}
+
+extension ParkingQueryFormViewController: UITableViewDelegate {
+    public func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
+        switch (query!.airportIATA, datePickerTag, indexPath.row) {
+        case (.none, .none, 0),
+             (.some, .none, 1),
+             (.none, .some(pickUpTag), 0),
+             (.some, .some(pickUpTag), 1):
+            // show dropoff date
+            datePickerTag = dropOffTag
+        case (.none, .none, 1),
+             (.some, .none, 2),
+             (.none, .some(dropOffTag), 2),
+             (.some, .some(dropOffTag), 3):
+            // show pickup date
+            datePickerTag = pickUpTag
+        default:
+            datePickerTag = nil
+        }
+
+        tableView.reloadData()
+        tableView.deselectRow(at: indexPath, animated: true)
+    }
+}
+
+extension ParkingQueryFormViewController: StringCellDelegate {
+    func stringCellValueDidChange(_ cell: StringCell) {
+        query?.airportIATA = cell.textField.text?.uppercased()
+        query?.boundingBox = nil
+    }
+}
+
+extension ParkingQueryFormViewController: DatePickerCellDelegate {
+    func datePickerCellValueDidChange(_ cell: DatePickerCell) {
+        guard let indexPath = tableView.indexPath(for: cell) else {
+            return
+        }
+
+        switch cell.tag {
+        case dropOffTag:
+            query!.dateRange = ClosedRange(uncheckedBounds: (lower: cell.datePicker.date, upper: query!.dateRange.upperBound))
+        case pickUpTag:
+            query!.dateRange = ClosedRange(uncheckedBounds: (lower: query!.dateRange.lowerBound, upper: cell.datePicker.date))
+        default:
+            Log("Unknown date cell", data: cell, level: .error)
+            break
+        }
+
+        let valueIndexPath = IndexPath(row: indexPath.row - 1, section: indexPath.section)
+        tableView.reloadRows(at: [valueIndexPath], with: .none)
+    }
+}
+
+extension ParkingQueryFormViewController: ParkingFilterContextObserving {
+    public func parkingFilterContextDidChangeSelectedFilter(_ context: ParkingFilterContext) {
+        query?.airportIATA = context.selectedFilter == 1 ? "" : nil
+
+        switch context.selectedFilter {
+        case nearMe:
```

Comment:
If the variables are for `nearMe` and `airport` are just syntactic sugar, then move them to the class scope, and use integer value checks here. Otherwise enumerate and keep in global scope. Having integers defined like it is now in the global scope is messy and hard to read.